tlsutil/config.go

```diff
@@ -537,21 +537,12 @@ func (c *Configurator) VerifyIncomingRPC() bool {
 }
 
 // This function acquires a read lock because it reads from the config.
-func (c *Configurator) outgoingRPCTLSDisabled() bool {
+func (c *Configurator) outgoingRPCTLSEnabled() bool {
 	c.lock.RLock()
 	defer c.lock.RUnlock()
 
-	// if AutoEncrypt enabled, always use TLS
-	if c.base.AutoTLS {
-		return false
-	}
-
-	// if CAs are provided or VerifyOutgoing is set, use TLS
-	if c.base.VerifyOutgoing {
-		return false
-	}
```

Comment:
This comment did not match the logic. I'm not sure if we should be looking at `c.caPool` here or not. But I updated the comment to match the code, assuming the code is right.

Reply:
yeah the code is correct. `VerifyOutgoing` decides what is going on. It is important to be able to load a ca without enabling outgoing TLS for the transition from unencrypted to encrypted communication. Which is why the comment is incorrect.